user_modules/dummy_connecttest.lua

```diff
@@ -371,37 +371,42 @@ end
 --! @example: self:initHMI_onReady(local_hmi_table) ]]
 function module:initHMI_onReady(hmi_table)
   local exp_waiter = commonFunctions:createMultipleExpectationsWaiter(module, "HMI on ready")
-
+  local delayValue = 0
   local function ExpectRequest(name, hmi_table_element)
-    if hmi_table_element.occurrence == 0 then
-      EXPECT_HMICALL(name, hmi_table_element.params)
-      :Times(0)
-      commonTestCases:DelayedExp(3000)
-      return
-    end
     local event = events.Event()
     event.level = 2
     event.matches = function(self, data)
       return data.method == name
     end
     local exp = EXPECT_HMIEVENT(event, name)
-    :Times(hmi_table_element.mandatory and 1 or AnyNumber())
+    :Times(hmi_table_element.occurrence or hmi_table_element.mandatory and 1 or AnyNumber())
     :Do(function(_, data)
+      if hmi_table_element.occurrence ~= 0 and hmi_table_element.params then
         xmlReporter.AddMessage("hmi_connection","SendResponse",
         {
           ["methodName"] = tostring(name),
           ["mandatory"] = hmi_table_element.mandatory,
           ["params"] = hmi_table_element.params
         })
         self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", hmi_table_element.params)
-      end)
-    if hmi_table_element.mandatory then
-      exp_waiter:AddExpectation(exp)
+      end
+    end)
+
+    if hmi_table_element.wait and delayValue < hmi_table_element.wait then
```

Comment:
@aderiabin This logic won't work in case:
occurence > 0 and wait = 11000
Since default timeout of 10s will be used for :Timeout() function and event won't be caught if it's sent between 10s and 11s

Reply:
@dboltovskyi Fixed in next manner (see afd93d6):
```
    if hmi_table_element.occurrence ~= 0 then
```
Only in case `occurrence` not equals 0 make sense add Timeout, pin expectation and register expectation in multiexpectation.
```
      if hmi_table_element.wait then
        exp = exp:Timeout(hmi_table_element.wait)
      end

      if hmi_table_element.mandatory then
      exp_waiter:AddExpectation(exp)
      end

      if hmi_table_element.pinned then
        exp:Pin()
      end
```
Only in case `occurrence` equals 0 make sense create DelayedExp.
```  
  else
      local waitTime = hmi_table_element.wait or 0
      if delayValue < waitTime then
        delayValue = waitTime
        commonTestCases:DelayedExp(delayValue)
      end
    end
```
Please take attention on
`exp:Timeout(hmi_table_element.wait)` -- set timeout for current expectation as is
but 
`DelayedExp(delayValue)` -- max of existed `wait` parameters (does not create expectation in case exists one with bigger delay)